urllib3/exceptions.py

```diff
@@ -196,6 +196,14 @@ class ResponseNotChunked(ProtocolError, ValueError):
     pass
 
 
+class BodyNotHttplibCompatible(AssertionError, ValueError):
```

Comment:
Hrm. I understand that concern, but I think that frankly we have a class defined as the base exception for the library, and that we should use it as much as possible rather than add a new backward-compat version. The name isn't ideal, but I'm not too fussed about that tbh. =(